src/Sarif/TypedPropertiesDictionary.cs

```diff
@@ -19,7 +19,7 @@ public TypedPropertiesDictionary() : this(null, StringComparer.Ordinal)
         {
         }
 
-        public TypedPropertiesDictionary(PropertiesDictionary initializer, IEqualityComparer<string> comparer) : base(comparer)
+        public TypedPropertiesDictionary(PropertiesDictionary initializer, IEqualityComparer<string> comparer) : base(comparer ?? StringComparer.Ordinal)
```

Comment:
>StringComparer [](http://example.com/codeflow?start=130&length=14)

As I mentioned in the description, when this is instantiated by the SarifWorkItemContext, the comparer will be null.

So, to use the same strategy as the constructor above, I'm defaulting the comparer to StringComparer.Ordinal.

Reply:
Yes! Good fix. Did you confirm that this is the behavior for Dictionary<string, T> by examining the reference implementation? Ideally, we would match the behavior of that concrete type.